frontend/src/App.tsx

```diff
@@ -147,30 +170,13 @@ function App() {
             horizontal: 'left',
           }}
         >
-          <Router>
-            <div className={classes.root}>
-              <CssBaseline />
-              <AppBar
-                position="fixed"
-                className={classes.appBar}
-                elevation={1}
-              >
-                <Toolbar>
-                  <div style={{flex: '1 0 0'}} />
-                  <ClusterTitle />
-                  <div style={{flex: '1 0 0'}} />
-                </Toolbar>
-              </AppBar>
-              <Sidebar />
-              <main className={classes.content}>
-                <div className={classes.toolbar} />
-                <Container maxWidth="lg">
-                  <RouteSwitcher />
-                </Container>
-              </main>
-              <ActionsNotifier />
-            </div>
-          </Router>
+          {isElectron() ?
+            <HashRouter>
+              {routerChildren}
+            </HashRouter> :
+            <BrowserRouter>
+              {routerChildren}
+            </BrowserRouter>}
```

Comment:
You could create a `Router` var/component that depending on the value of the `isElectron()` would return one router or another.
Then the changes are minimal.

Reply:
Yeah that would be more elegant thanks!!